src/redirect_resources.rs

```diff
@@ -0,0 +1,101 @@
+use std::collections::HashMap;
+use serde::{Deserialize, Serialize};
+use crate::resources::{Resource, ResourceType};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct RedirectResource {
+    pub content_type: String,
+    pub data: String
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default)]
+pub struct RedirectResourceStorage {
+    pub resources: HashMap<String, RedirectResource>,
+    #[serde(default)]
+    pub aliases: HashMap<String, String>,
+}
+
+impl RedirectResourceStorage {
+    pub fn from_resources(resources: &[Resource]) -> Self {
+        let mut redirectable_resources: HashMap<String, RedirectResource> = HashMap::new();
+        let mut aliases: HashMap<String, String> = HashMap::new();
+
+        resources.iter().filter_map(|descriptor| {
+            if let ResourceType::Mime(ref content_type) = descriptor.kind {
+                let resource = RedirectResource {
+                    content_type: content_type.clone().into(),
+                    data: descriptor.content.to_owned(),
+                };
+                Some((descriptor.name.to_owned(), descriptor.aliases.to_owned(), resource))
+            } else {
+                None
+            }
+        }).for_each(|(name, res_aliases, resource)| {
+            redirectable_resources.insert(name.clone(), resource);
+            res_aliases.iter().for_each(|alias| {
+                aliases.insert(alias.to_owned(), name.to_owned());
+            });
+        });
+
+        Self {
+            resources: redirectable_resources,
+            aliases,
+        }
+    }
+
+    pub fn get_resource(&self, name: &str) -> Option<&RedirectResource> {
+        self.resources.get(name).or_else(|| self.aliases.get(name).and_then(|name| self.resources.get(name)))
```

Comment:
IMO this would be better handled with ref-counted pointers to avoid additional copies and a two-step lookup

Reply:
My only reason for avoiding that was that `serde` doesn't really handle `Rc` correctly. The default implementation, which requires a `cfg` flag, ends up just cloning the data every time.

The primary name is generally the shortest one, so the second lookup should be relatively uncommon anyways.

It would be possible to get it working with some added logic (e.g. custom serializer / deserializer). Would you say it's worth the additional complexity?